feat: add `stats/incr/nanewstdev`

Girish-Garg opened the pull request:
Resolves #5557.

## Description

> What is the purpose of this pull request?

This pull request:

-   add incremental exponentially weighted standard deviation accumulator by making wrapper over `@stdlib/stats/incr/ewstdev` main function

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5557 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Test:
![image](https://github.com/user-attachments/assets/d3976587-9679-4cca-a6af-7008f420008d)

Examples:
![image](https://github.com/user-attachments/assets/51b9b8d6-dfd9-4349-8872-738580f17dfc)

Benchmark:
![image](https://github.com/user-attachments/assets/313db8a4-2141-4f22-8241-bfef1ba741c6)

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Girish-Garg committed:
feat: add incremental exponentially weighted standard deviation accumulator

Girish-Garg commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Girish-Garg committed:
chore: fixed linting and clean-up

Girish-Garg committed:
chore: fixed linting

Girish-Garg commented:
@kgryte i think this pr is good enough to merge, if there are any changes required let me know

Girish-Garg committed:
chore: fix example file

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanewstdev/package.json:
```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@stdlib/stats/incr/nanewstdev",
+  "version": "0.0.0",
+  "description": "Compute an exponentially weighted standard deviation incrementally, ignoring `NaN` values.",
```
Don't wrap `NaN` in backticks in `package.json` descriptions.

```suggestion
  "description": "Compute an exponentially weighted standard deviation incrementally, ignoring NaN values.",
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanewstdev/README.md:
```diff
@@ -0,0 +1,171 @@
+
+## Notes
+
+-   Input values are **not** type checked.If non-numeric inputs are possible, but you are advised to type check and handle accordingly **before** passing the value to the accumulator function.
```
Missing space after the period, and the "but" makes the sentence ungrammatical. Remove "but".

```suggestion
-   Input values are **not** type checked. If non-numeric inputs are possible, you are advised to type check and handle accordingly **before** passing the value to the accumulator function.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanewstdev/lib/main.js:
```diff
@@ -0,0 +1,79 @@
+function incrnanewstdev( alpha ) {
+	var ewstdev;
+
+	ewstdev = increwstdev(alpha);
```
Missing spaces inside parentheses.

```suggestion
	ewstdev = increwstdev( alpha );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanewstdev/examples/index.js:
```diff
@@ -0,0 +1,37 @@
+for ( i = 0; i < 100; i++ ) {
+	v = ( randu() < 0.1 ) ? NaN : randu() * 100.0; // 10% chance of a NaN
+	s = accumulator( v );
+	console.log( '%d\t%d', v.toFixed( 4 ), ( s === null ) ? NaN : s.toFixed( 4 ) );
```
`.toFixed()` returns a string. Use `%s`, not `%d`.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanewstdev/README.md:
```diff
@@ -0,0 +1,171 @@
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/stats/incr/ewvariance`][@stdlib/stats/incr/ewvariance]</span><span class="delimiter">: </span><span class="description">compute an exponentially weighted variance incrementally.</span>
+-   <span class="package-name">[`@stdlib/stats/incr/mstdev`][@stdlib/stats/incr/mstdev]</span><span class="delimiter">: </span><span class="description">compute a moving corrected sample standard deviation incrementally.</span>
+-   <span class="package-name">[`@stdlib/stats/incr/stdev`][@stdlib/stats/incr/stdev]</span><span class="delimiter">: </span><span class="description">compute a corrected sample standard deviation incrementally.</span>
```
Can remove as auto-generated.

Planeshifter committed:
chore: address review feedback and update copyright years

- remove backticks around NaN in package.json description
- fix grammar in README notes
- add spaces inside parentheses in main.js
- use %s format specifier for toFixed string output in example
- remove manually populated See Also section
- use `format` for benchmark name string concatenation

Planeshifter committed:
fix: correct function name and provide required `alpha` argument in doctest

The `@module` `@example` used the wrong function name and called the
accumulator factory without the required smoothing factor, causing the
jsdoc-doctest lint to fail.

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/nanewstdev

Planeshifter committed:
chore: address review feedback

- remove unused `isnan` import from README example
- add spaces inside parentheses to satisfy `space-in-parens`
